### DISCUSSION TITLE
Queue Url differs from SQSClient endpoint message

### DISCUSSION THREAD

**Discussion by jmckeeext** [bbc/sqs-consumer#517] [Help] [answered]
### Summary

I'm using LocalStack and have sqs-consumer pointed to my LocalStack queue for local dev work but I repeatedly get message 'QueueUrl=http://sqs.us-east-1.localhost.localstack.cloud:4566/000000000000/localstack-queue differs from SQSClient resolved endpoint=https://sqs.us-east-1.amazonaws.com/, using QueueUrl host as endpoint.'
in terminal window. 

Is there a way to stop this message without having to custom configure an SQSClient?

**Comment by jagiles-ren** [comment, marked as answer]
@jmckeeext - I recently ran into this same issue. It actually isn't coming from sqs-consumer, but instead from the AWS SDK (see: https://github.com/aws/aws-sdk-js-v3/blob/0ade74b1adcf33f36078454ce3414408afe3813b/packages/middleware-sdk-sqs/src/queue-url.ts#L66)

What I did to resolve this error myself but still be able to use localstack was to do something similar to what the SDK is doing, and resolving the endpoint from the origin of the queue URL. That way, if the Queue URL is a localstack URL, it will resolve the endpoint for that client. Basically doing the same thing as `useQueueUrlAsEndpoint` is/was doing under the covers. I suspect AWS may be deprecating this functionality in the future, which is why the warning is presented (I have no basis for this assumption).

Example:
```typescript
import { SQSClient } from '@aws-sdk/client-sqs';
import { Consumer } from 'sqs-consumer';

// Purely for demonstration... obviously these values should come from configuration
const queueUrl = 'http://sqs.us-east-1.localhost.localstack.cloud:4566/000000000000/some-queue';

const consumer = new Consumer({
  sqs: new SQSClient({
    endpoint: new URL(queueUrl).origin,
  }),
  queueUrl,
  // Other SQS consumer related properties here...
});
```

**Reply by jmckeeext** [reply]
Thanks. I thought perhaps I was using sqs-consumer incorrectly. did not realize the message was coming from aws sdk.

**Comment by nicholasgriffintn** [comment]
Hey sorry, I didn't see this, my bad, I think usequeueurlasendpoint should be the resolution?

**Reply by jagiles-ren** [reply]
It unfortunately is not in more recent versions of the AWS SDK (v3). That warning message is displayed despite setting `useQueueUrlAsEndpoint = true`. That setting is actually what caused me to start seeing the warning from the SDK.

**Reply by nicholasgriffintn** [reply]
Okay, well we can't really provide support for the AWS SDK really to be honest and those errors are coming from them, not working for AWS, we can't provide support for their products.

**Reply by jagiles-ren** [reply]
I wasn't assuming you would! 😉 I was simply providing an alternative approach to configuring the `SQSClient` provided to the `Consumer` such that the warning does not appear from the aws-sdk. Since I found this discussion during a web search, I assumed others may as well and wanted to provide direction on how to address the concern that led me (and I suspect others) to this discussion thread.

Looking at the source code for the aws-sdk, I strongly suspect that this message will always appear when using the `useQueueUrlAsEndpoint` option along with a localstack URL, since the aws-sdk is now detecting when the discovered endpoint and the Queue URL have different origins and displays this warning message.
